modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/option/DocumentLibraryCommerceOptionTypeImpl.java

```diff
@@ -0,0 +1,172 @@
+import com.liferay.headless.commerce.delivery.catalog.dto.v1_0.ProductOption;
+import com.liferay.portal.kernel.language.Language;
+import com.liferay.portal.kernel.theme.PortletDisplay;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.HashMapBuilder;
+import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portal.kernel.util.WebKeys;
+import com.liferay.portal.template.react.renderer.ComponentDescriptor;
+import com.liferay.portal.template.react.renderer.ReactRenderer;
+import com.liferay.portal.vulcan.dto.converter.DTOConverter;
+import com.liferay.portal.vulcan.dto.converter.DTOConverterRegistry;
+import com.liferay.portal.vulcan.dto.converter.DefaultDTOConverterContext;
+
+import java.io.PrintWriter;
+
+import java.util.Locale;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Andrea Sbarra
+ */
+@Component(
+	property = {
+		"commerce.option.type.display.order:Integer=200",
+		"commerce.option.type.key=" + CPConstants.PRODUCT_OPTION_DOCUMENT_LIBRARY_KEY
+	},
+	service = CommerceOptionType.class
+)
+public class DocumentLibraryCommerceOptionTypeImpl
+	implements CommerceOptionType {
+
+	@Override
+	public String getKey() {
+		return CPConstants.PRODUCT_OPTION_DOCUMENT_LIBRARY_KEY;
+	}
+
+	@Override
+	public String getLabel(Locale locale) {
+		return _language.get(locale, "upload");
+	}
+
+	@Override
+	public boolean hasValues() {
+		return false;
+	}
+
+	@Override
+	public void render(
+			CPDefinitionOptionRel cpDefinitionOptionRel,
+			long defaultCPInstanceId, boolean forceRequired, String json,
+			HttpServletRequest httpServletRequest,
+			HttpServletResponse httpServletResponse)
+		throws Exception {
+
+		if (cpDefinitionOptionRel == null) {
+			return;
+		}
+
+		PrintWriter printWriter = httpServletResponse.getWriter();
+
+		ThemeDisplay themeDisplay =
+			(ThemeDisplay)httpServletRequest.getAttribute(
+				WebKeys.THEME_DISPLAY);
+
+		PortletDisplay portletDisplay = themeDisplay.getPortletDisplay();
+
+		DDMFormField ddmFormField = new DDMFormField(
+			cpDefinitionOptionRel.getKey(),
+			cpDefinitionOptionRel.getCommerceOptionTypeKey());
+
+		Locale locale = _portal.getLocale(httpServletRequest);
+
+		LocalizedValue ddmFormFieldLabelLocalizedValue = new LocalizedValue(
+			locale);
+
+		ddmFormFieldLabelLocalizedValue.addString(
+			locale, cpDefinitionOptionRel.getName(locale));
+
+		ddmFormField.setLabel(ddmFormFieldLabelLocalizedValue);
+
+		ddmFormField.setName(cpDefinitionOptionRel.getKey());
+		ddmFormField.setRequired(cpDefinitionOptionRel.isRequired());
+
+		DDMForm ddmForm = new DDMForm();
+
+		ddmForm.addAvailableLocale(locale);
+		ddmForm.addDDMFormField(ddmFormField);
+		ddmForm.setDefaultLocale(locale);
```

Comment:
not sure about the order here as is not "as used"
I think it should be 
```
DDMForm ddmForm = new DDMForm
ddmForm.addAvailableLocale(locale);

logic to create the ddmFormField
ddmForm.addDDMFormField(ddmFormField);

ddmForm.setDefaultLocale(locale);
```

the portletDisplay is used only on line 120, so the get of it should be done just before it.
But probably this is a copy / paste from somewhere else so ... 🤞